test/new-e2e/utils/e2e/client/agent_cmd_runner.go

```diff
@@ -0,0 +1,89 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package client
+
+import (
+	"errors"
+	"regexp"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/cenkalti/backoff"
+)
+
+// AgentCommandRunner provides methods to run commands on the Agent.
+type AgentCommandRunner struct {
+	cmdRunner *vmClient
+	os        os.OS
+}
+
+// Create a new instance of Agent
+func newAgentCommandRunner(client *vmClient, os os.OS) *AgentCommandRunner {
+	agent := &AgentCommandRunner{
+		cmdRunner: client, os: os,
+	}
+	agent.waitForReadyTimeout(1 * time.Minute)
+	return agent
+}
+
+func (agent *AgentCommandRunner) getCommand(parameters string) string {
+	return agent.os.GetRunAgentCmd(parameters)
+}
+
+func (agent *AgentCommandRunner) executeCommand(command string, commandArgs ...AgentArgsOption) string {
+	args := newAgentArgs(commandArgs...)
+	return agent.cmdRunner.Execute(agent.getCommand(command) + " " + args.Args)
+}
+
+func (agent *AgentCommandRunner) Version(commandArgs ...AgentArgsOption) string {
+	return agent.executeCommand("version", commandArgs...)
+}
+
+func (agent *AgentCommandRunner) Config(commandArgs ...AgentArgsOption) string {
+	return agent.executeCommand("config", commandArgs...)
+}
+
+type Status struct {
+	Content string
+}
+
+func newStatus(s string) *Status {
+	return &Status{Content: s}
+}
+
+// isReady true if status contains a valid version
+func (s *Status) isReady() (bool, error) {
```

Comment:
In most cases, the user want the agent to be ready before running a command. That's explain why I did this change.

To answer your question, instead of calling `waitForReadyTimeout` in `newAgentCommandRunner`, I can call `waitForReadyTimeout` lazily on the first command of the Agent and export `isReady`. 
Please note that `waitForReady` would still not be public as it doesn't seem to be needed in the case you described above.

If you want to keep `waitForReady` public, then we have to find a name which make it obvious that the user doesn't have to call it before doing anything on the Agent. What do you think?

Reply:
Should we add the `(a *Agent) IsReady()` public function here